pages/add/add.jsx

```diff
@@ -1,12 +1,161 @@
 import React from 'react';
+import { browserHistory } from 'react-router';
+import HintMessage from '../../components/hint-message/hint-message.jsx';
+import user from '../../js/app-user';
+import env from "../../config/env.generated.json";
+
+const CreatorInputField = () => {
+  return (
+    <input name="creators" type="text" placeholder="Name" width="150" className="form-control" />
+  );
+};
 
 export default React.createClass({
+  getInitialState() {
+    return {
+      numCreatorFields: 1,
+      user
+    };
+  },
+  componentDidMount() {
+    // let's verify what the logged in status is
+    user.verify(this.props.router.location, () => this.setState({ user }));
+  },
+  handleSignInBtnClick(event) {
+    event.preventDefault();
+
+    let redirectUrl = `${env.ORIGIN}${this.props.router.getCurrentLocation().pathname}`;
+    user.login(redirectUrl);
+  },
+  handleLogOutBtnClick(event) {
+    event.preventDefault();
+
+    user.logout();
+    browserHistory.push({
+      pathname: `/featured`
+    });
+  },
+  handleCreatorClick(event) {
+    event.preventDefault();
+    this.setState({numCreatorFields: this.state.numCreatorFields+1});
+  },
   render() {
+    let creatorFields = [];
+
+    for (var i=0; i<this.state.numCreatorFields; i++) {
+      creatorFields.push( <CreatorInputField key={i} />);
+    }
+
+    let redirectUrl = `${env.ORIGIN}${this.props.router.getCurrentLocation().pathname}`;
+    let anonymousContent = (<HintMessage imgSrc={`/assets/svg/icon-user.svg`}
+                                         header={`Please sign in to add a post`}
+                                         externalLink={user.getLoginURL(redirectUrl)}
+                                         linkText={`Sign in`}
+                                         onClick={this.handleSignInBtnClick}>
+                            <p>Please note, only Mozilla staff can login now as we test this new platform.</p>
+                          </HintMessage>);
+
+
+    let failurePrompt= ( <HintMessage imgSrc={`/assets/svg/icon-user.svg`}
+                                      header={`Sign in failed`}
+                                      internalLink={`/featured`}
+                                      linkText={`Explore featured`}>
+                          <p>Only Mozilla staff can login now as we test this new platform. Check back soon!</p>
+                        </HintMessage>);
+
+    let contentForLoggedInUser = (
+      <div>
+        <h1>Share with the Network</h1>
+        <p>Do you have something to share? If it might be useful to someone in our network, share it here! Pulse includes links to products and software tools, research reports and findings, think pieces, white papers, interviews, and curricula. If it might be useful, share it … at any stage or fidelity.</p>
+        <form action="" method="post">
```

Comment:
Oh yea the form part is definitely not finished yet. I wanted to get login logics working first. Maybe I should remove the form chunks from this PR and create followup PR to take care it separately?